src/seedu/addressbook/data/AddressBook.java

```diff
@@ -72,7 +72,8 @@ public UniquePersonList getAllPersons() {
      * Returns a new UniquePersonList of all persons in the address book at the time of the call sorted in alphabetical order.
      */
     public UniquePersonList getSortedAllPersons() {
-        return new UniquePersonList(allPersons).sort();
+        UniquePersonList toSort = new UniquePersonList(allPersons);
+        return toSort.sort();
```

Comment:
Doesn't update the main list of persons? Only sort the new list created (the copy of the main list)?